packages/core/src/components/section/section.tsx

```diff
@@ -74,6 +74,7 @@ export interface SectionProps extends Props, Omit<HTMLDivProps, "title">, React.
 
     /**
      * Element to render on the right side of the section header.
+     * Note that the header will only be rendered if `title` is provided.
```

Comment:
Nice catch copying this comment over from the other props that are no-ops if `title` is not provided.